classes/post-form.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * Class Post_Form
+ */
+class Post_Form {
+	/**
+     *  Build template for client
+	 *
+	 * @param array{
+	 *     title: string,
+	 *     description: string,
+     *     show_categories: string,
+	 *     not_registered: string,
+	 * } $settings
+	 *
+	 * @since 1.0.0
+	 *  @access public
+	 *  @static
+	 *
+	 * @return string
+	 */
+	static function get_render_form( array $settings ): string {
+		$user = wp_get_current_user();
+        ob_start(); ?>
+			<section class="el-post-form-widget">
+                <h3 class="el-post-form-head"><?php esc_html_e($settings['title']); ?></h3>
+				<?php if ($user->exists()): ?>
+                    <p class="el-post-form-desc"><?php esc_html_e($settings['description']); ?></p>
```

Comment:
I'd say that if we escape dynamic data and don't need the domain -- we should use `echo esc_*()` functions.